output/src/main/java/com/scottlogic/deg/output/writer/sql/SqlDataSetWriter.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2019 Scott Logic Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.scottlogic.deg.output.writer.sql;
+
+import com.scottlogic.deg.common.output.GeneratedObject;
+import com.scottlogic.deg.common.profile.Field;
+import com.scottlogic.deg.common.profile.ProfileFields;
+import com.scottlogic.deg.output.writer.DataSetWriter;
+
+import java.io.IOException;
+import java.io.Writer;
+import java.math.BigDecimal;
+import java.time.OffsetDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+class SqlDataSetWriter implements DataSetWriter {
+    private static final DateTimeFormatter standardDateFormat = DateTimeFormatter.ISO_OFFSET_DATE_TIME;
+    private final Writer writer;
+    private final ProfileFields fieldOrder;
+    private final String tableName;
+
+    SqlDataSetWriter(Writer writer, ProfileFields fieldOrder, String tableName) {
+        this.writer = writer;
+        this.fieldOrder = fieldOrder;
+        this.tableName = tableName;
+    }
+
+    @Override
+    public void writeRow(GeneratedObject row) throws IOException {
+        // eg insert into dbo.Currencies (currency, from_date, to_date) values
+        //    ( 'YYY', '1762-02-16T00:00:00Z', '1980-02-16T00:00:00Z');
+
+        Optional<String> columnNamesCsvString = reduceStringStreamToCsvString(
+            fieldOrder.stream()
+                .map(Field::toString)
```

Comment:
I'm generally not a fan of using `toString` for logic in Java, as now our implementation of `toString` is tied to our logic, rather than being malleable as necessary to aid debugging etc.. I believe this information should be exposed by a different method on the `Field` class.

Reply:
added name() method that calling now